src/hugr/rewrite/insert_identity.rs

```diff
@@ -53,11 +57,15 @@ impl Rewrite for IdentityInsertion {
         Conditions:
         1. post_port is Value kind.
         2. post_port is connected to a sibling of post_node.
+        3. post_port is input.
          */
 
         unimplemented!()
     }
     fn apply(self, h: &mut Hugr) -> Result<Self::ApplyResult, IdentityInsertionError> {
+        if self.post_port.direction() != Direction::Incoming {
+            return Err(IdentityInsertionError::PortIsOutput);
+        }
```

Comment:
Would it now be safe to use `unwrap()` at line 72 as otherwise the hugr is invalid? Or is there an edge case I've forgotten?